framework/CustomModes/MPISimulationMode.py

```diff
@@ -0,0 +1,195 @@
+
+  def __createAndRunQSUB(self, runInfoDict):
+    """
+      Generates a PBS qsub command to run the simulation
+      @ In, runInfoDict, dict, dictionary of run info.
+      @ Out, remoteRunCommand, dict, dictionary of command.
+    """
+    # Check if the simulation has been run in PBS mode and, in case, construct the proper command
+    #while true, this is not the number that we want to select
+    if self.__coresNeeded is not None:
+      coresNeeded = self.__coresNeeded
+    else:
+      coresNeeded = runInfoDict['batchSize']*runInfoDict['NumMPI']
+    if self.__memNeeded is not None:
+      memString = ":mem="+self.__memNeeded
+    else:
+      memString = ""
+    #batchSize = runInfoDict['batchSize']
+    frameworkDir = runInfoDict["FrameworkDir"]
+    ncpus = runInfoDict['NumThreads']
+    jobName = runInfoDict['JobName'] if 'JobName' in runInfoDict.keys() else 'raven_qsub'
+    #check invalid characters
+    validChars = set(string.ascii_letters).union(set(string.digits)).union(set('-_'))
+    if any(char not in validChars for char in jobName):
+      raise IOError('JobName can only contain alphanumeric and "_", "-" characters! Received'+jobName)
+    #check jobName for length
+    if len(jobName) > 15:
+      jobName = jobName[:10]+'-'+jobName[-4:]
+      print('JobName is limited to 15 characters; truncating to '+jobName)
+    #Generate the qsub command needed to run input
+    command = ["qsub","-N",jobName]+\
+              runInfoDict["clusterParameters"]+\
+              ["-l",
+               "select="+str(coresNeeded)+":ncpus="+str(ncpus)+":mpiprocs=1"+memString,
+               "-l","walltime="+runInfoDict["expectedTime"],
+               "-l","place="+self.__place,"-v",
+               'COMMAND="../raven_framework '+
+               " ".join(runInfoDict["SimulationFiles"])+'"',
+               runInfoDict['RemoteRunCommand']]
+    #Change to frameworkDir so we find raven_qsub_command.sh
+    remoteRunCommand = {}
+    remoteRunCommand["cwd"] = frameworkDir
+    remoteRunCommand["args"] = command
+    print("remoteRunCommand",remoteRunCommand)
+    return remoteRunCommand
+
+  def remoteRunCommand(self, runInfoDict):
+    """
+      If this returns None, don't do anything.  If it returns a
+      dictionary, then run the command in the dictionary.
+      @ In, runInfoDict, dict, the run info dictionary
+      @ Out, remoteRunCommand, dict, a dictionary with information for running.
+    """
+    if not self.__runQsub or self.__inPbs:
+      return None
+    assert self.__runQsub and not self.__inPbs
+    return self.__createAndRunQSUB(runInfoDict)
+
+  def XMLread(self, xmlNode):
+    """
+      XMLread is called with the mode node, and is used here to
+      get extra parameters needed for the simulation mode MPI.
+      @ In, xmlNode, xml.etree.ElementTree.Element, the xml node that belongs to this class instance
+      @ Out, None
+    """
+    for child in xmlNode:
+      if child.tag == "nodefileenv":
```

Comment:
Well, looking again, I left the lower() on runqsub (otherwise it would be another input change), but for the rest I don't force it to lower case.